### ISSUE TITLE
Reading and writing YAML documents internally should preserve custom constructors

### ISSUE THREAD

**Issue by jeremydw** [grow/grow#806]
Currently if you have a YAML document like the below:

```
$localization:
  locales: !g.yaml /content/products/phone.yaml?locales
foo: bar
```

Reading it and then writing it using `pod.write_yaml` causes  `!g.yaml /content/products/phone.yaml?locales` to be expanded and written. The custom tag should be preserved when written.

**Comment by Zoramite** [comment]
We would need to add a custom object as the return from the constructor that understands how to be dumped correctly. Similar to the plain text dumper: https://github.com/grow/grow/blob/master/grow/common/yaml_utils.py

We could actually test the plain text yaml util to see if it meets the need. I've had to use it in other places when I need to preserve the constructor output.

**Comment by Zoramite** [comment]
I believe this is fixed with #809
